core/src/client/mod.rs

```diff
@@ -1310,15 +1312,17 @@ impl KakarotClient for KakarotClientImpl {
         blockhash_opt: Option<H256>,
         blocknum_opt: Option<U256>,
     ) -> Result<BlockTransactions, KakarotClientError> {
-        let mut transactions_vec = vec![];
-        for transaction in initial_transactions {
-            let tx_value = self
-                .starknet_tx_into_kakarot_tx(transaction, blockhash_opt, blocknum_opt)
-                .await;
-            if let Ok(val) = tx_value {
-                transactions_vec.push(val)
-            }
-        }
+        // TODO replace filter_map to handle errors
```

Comment:
why is there a TODO here?

Reply:
Errors were not handled previously but maybe we should I'm not sure

Reply:
I think they're handled in the filter_map, no?

Reply:
Ahh okay I misunderstood, you said, to replace the filter_map with actual error handling!
I think we don't need to do that, we might want to refactor the fact that `starknet_tx_into_kakarot_tx` returns a result instead of an option (would be better imo)

Let's remove this comment and ready to merge:)!

Reply:
I'm only using filter_map to remove them from the vector as it was done here: https://github.com/sayajin-labs/kakarot-rpc/blob/ab59ebfb3878b6606e293f7ce42150727cc973bb/core/src/client/mod.rs#L1318

Errors were not pushed in transactions_vec. I reproduced this behavior but I'm not sure if that's what we should do

Reply:
imho it's fine:)!

Reply:
Removed the comment !